manifests/init.pp

```diff
@@ -51,73 +51,48 @@
 # Copyright 2013 Thomas Linkin, unless otherwise noted.
 #
 class domain_membership (
-  $domain,
-  $username,
-  $password,
-  $secure_password = false,
-  $machine_ou      = undef,
-  $resetpw         = true,
-  $reboot          = true,
-  $reboot_apply    = 'finished',
-  $join_options    = '1',
-  $user_domain     = undef,
+  Stdlib::Fqdn $domain,
+  String $username,
+  String $password,
```

Comment:
Limiting this to a `String` will prevent passing a password that is already of type `Sensitive`.

For example, this would fail:

```puppet
class { 'domain_membership':
  password => Sensitive('foo'),
}
```

This should probably be `Variant[String,Sensitive]`